itou/users/models.py

```diff
@@ -1027,6 +1036,9 @@ def save(self, force_insert=False, force_update=False, using=None, update_fields
             if update_fields is not None:
                 update_fields = set(update_fields) | {"pe_obfuscated_nir", "pe_last_certification_attempt_at"}
         super().save(force_insert=force_insert, force_update=force_update, using=using, update_fields=update_fields)
+        if not self.asp_uid:
+            self.asp_uid = salted_hmac(key_salt="job_seeker.id", value=self.user_id).hexdigest()[:30]
```

Comment:
Je pense que ta version fonctionne vu qu'on fait le `save()` avant mais est-ce qu'il y a un moment où `user_id` pourrais ne pas être synchro avec `user.pk` ? Je pense au cas où on changerais l'utilisateur, ce qui ne devrais jamais arriver mais...
```suggestion
            self.asp_uid = salted_hmac(key_salt="job_seeker.id", value=self.user.pk).hexdigest()[:30]
```

Reply:
J'ai l'impression que l'argument fonctionne dans les deux sens si l'on fait:
```
profile.user = mon_objet_django
profile.save()
```
ou
```
profile.user_id = mon_objet_django.pk
profile.save()
```
Et j'ai l'impression que Django fait bien le boulot, dans les deux sens:
```
In [17]: pro.validated_by
Out[17]: <User: Pamela HOLDER — email9@domain.com>

In [18]: pro.validated_by_id
Out[18]: 55

In [19]: pro.validated_by = pro3.validated_by

In [20]: pro.validated_by
Out[20]: <User: Rebecca JOHNSON — email13@domain.com>

In [21]: pro.validated_by_id
Out[21]: 59
```
et
```
In [23]: pro.validated_by
Out[23]: <User: Rebecca JOHNSON — email13@domain.com>

In [24]: pro.validated_by_id
Out[24]: 59

In [25]: pro.validated_by_id = pro4.validated_by.pk

In [26]: pro.validated_by
Out[26]: <User: Thomas RODGERS — email17@domain.com>

In [27]: pro.validated_by_id
Out[27]: 63
```

Reply:
Merci pour le test, AJA j'ai envie de dire :)